fw/apm.h

```diff
@@ -59,41 +60,40 @@ static const unsigned int tfw_pstats_ith[] = {
 
 #define T_PSZ	_TFW_PSTATS_IDX_COUNT
 
-/*
- * Structures for health monitoring statistics accountings
- * in procfs.
- */
-typedef struct {
-	int		code;
-	unsigned int	sum;
-} TfwHMCodeStats;
+int tfw_apm_add_srv(TfwServer *srv);
+void tfw_apm_del_srv(TfwServer *srv);
 
 /*
- * @rtime	- time until next server health checking;
- * @ccnt	- count of @rsums elements;
- * @rsums	- array of counters for separate HTTP codes;
+ * All blocks of the procedures listed below, separated by spaces, are not
+ * data-coupled. It can be said that these are independent submodules of the
+ * APM module.
  */
-typedef struct {
-	unsigned int	rtime;
-	unsigned int	ccnt;
-	TfwHMCodeStats	*rsums;
-} TfwHMStats;
 
-int tfw_apm_add_srv(TfwServer *srv);
-void tfw_apm_del_srv(TfwServer *srv);
+/* Procedures related to statistics (avg/min/max/percentiles).
+ * Configured by the 'apm_stats' directive.
+ */
 void tfw_apm_update(void *apmref, unsigned long jtstamp, unsigned long jrtime);
 int tfw_apm_stats(void *apmref, TfwPrcntlStats *pstats);
-int tfw_apm_stats_bh(void *apmref, TfwPrcntlStats *pstats);
-int tfw_apm_pstats_verify(TfwPrcntlStats *pstats);
+/* Displayed in the perfstat, not in a backend statistics. */
+void tfw_apm_update_global(unsigned long jtstamp, unsigned long jrtime);
+int tfw_apm_stats_global(TfwPrcntlStats *pstats);
+
+/*
+ * Although these procedures are used for health monitoring, they also collect
+ * overall statistics on responses from each server. Therefore, if health
+ * monitoring is not active for a server, these procedures will still be
```

Comment:
`will still be executed` -> `are still executed`.

Reply:
Fixed